src/main/java/com/meilisearch/sdk/json/GsonJsonHandler.java

```diff
@@ -26,9 +29,20 @@ public String encode(Object o) throws MeilisearchException {
             return (String) o;
         }
         // TODO: review later
+        // This is a workaround to encode the Key class properly
         if (o != null && o.getClass() == Key.class) {
             GsonBuilder builder = new GsonBuilder();
-            this.gson = builder.serializeNulls().setDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'").create();
+            this.gson = builder.setDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'").create();
+
+            Key key = (Key) o;
+            if (key.getExpiresAt() == null) {
+                JsonElement jsonElement = gson.toJsonTree(o);
+                JsonObject jsonObject = jsonElement.getAsJsonObject();
+                jsonObject.add("expiresAt", JsonNull.INSTANCE);
+                o = jsonObject;
+                this.gson =
+                        builder.serializeNulls().setDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'").create();
```

Comment:
what would be the value of `expiresAt` if we enter this condition?

Reply:
the condition is:
`if (key.getExpiresAt() == null)`
So `expiredAt` should be null to enter in this condition and it should be encoded as null. That why the json builder is set like:
```java
            this.gson =
                        builder.serializeNulls().setDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'").create();
```
with `serializeNulls()`